source/material_model/diffusion_dislocation.cc

```diff
@@ -265,7 +215,7 @@ namespace aspect
               // We have given the user freedom to apply alternative bounds, because in diffusion-dominated
               // creep (where n_diff=1) viscosities are stress and strain-rate independent, so the calculation
               // of compositional field viscosities is consistent with any averaging scheme.
-              out.viscosities[i] = average_value(composition, composition_viscosities, viscosity_averaging);
+              out.viscosities[i] = average_value(volume_fractions, composition_viscosities, viscosity_averaging);
```

Comment:
was this a bug?

Reply:
No. There were 3 different versions of this function (I compared them all with diff before merging them into one), and this one happened to use mass fractions (the composition), and then call the `compute_volume_fractions()` function inside of `average_value()`. But as the volume_fractions had already been computed before, I could just use them here and remove the call to `compute_volume_fractions()` inside of `average_value()`. So the functionality is the same as before, but the function is now consistently the same across all material models.